source/common/upstream/cluster_factory_impl.cc

```diff
@@ -52,8 +52,8 @@ std::pair<ClusterSharedPtr, ThreadAwareLoadBalancerPtr> ClusterFactoryImplBase::
     case envoy::config::cluster::v3::Cluster::EDS:
       cluster_type = "envoy.cluster.eds";
       break;
-    default:
-      NOT_REACHED_GCOVR_EXCL_LINE;
+    case SENTINEL_VALUES_USED:
```

Comment:
Cool. Didn't realize it was this simple. How about also wrapping the case here, so it would reduce to just:

`PROTO_SENTINEL_GUARD` at the end of each switch.

Something like:

`#define PROTO_SENTINEL_GUARD case SENITNEL_VALUES_USED: PANIC_DUE_TO_SENITNEL_VALUE`

Reply:
Yep, this looks better. One nit is I'd rename it to `PANIC_ON_PROTO_ENUM_SENTINEL_VALUES` to make it a bit clearer that this is proto-related and we are handing specific cases there. If someone wants this for something else they can alias, but I expect that to be uncommon.